Mathlib/AlgebraicGeometry/Scheme.lean

```diff
@@ -0,0 +1,345 @@
+    @IsIso LocallyRingedSpace _ _ _ f :=
+  forgetToLocallyRingedSpace.map_isIso f
+#align algebraic_geometry.Scheme.is_LocallyRingedSpace_iso AlgebraicGeometry.Scheme.is_locallyRingedSpace_iso
+
+-- Porting note: need an extra instance here.
+instance {X Y : Scheme} (f : X ⟶ Y) [IsIso f] (U) : IsIso (f.val.c.app U) :=
+  haveI := PresheafedSpace.c_isIso_of_iso f.val
+  NatIso.isIso_app_of_isIso _ _
+
+@[simp]
+theorem inv_val_c_app {X Y : Scheme} (f : X ⟶ Y) [IsIso f] (U : Opens X.carrier) :
+    (inv f).val.c.app (op U) =
+      X.presheaf.map
+          (eqToHom <| by rw [IsIso.hom_inv_id]; ext1; rfl :
+              (Opens.map (f ≫ inv f).1.base).obj U ⟶ U).op ≫
+        inv (f.val.c.app (op <| (Opens.map _).obj U)) := by
+  rw [IsIso.eq_comp_inv]
+  erw [← Scheme.comp_val_c_app]
+  rw [Scheme.congr_app (IsIso.hom_inv_id f), Scheme.id_app, ← Functor.map_comp, eqToHom_trans,
+    eqToHom_op]
+#align algebraic_geometry.Scheme.inv_val_c_app AlgebraicGeometry.Scheme.inv_val_c_app
+
+/-- Given a morphism of schemes `f : X ⟶ Y`, and open sets `U ⊆ Y`, `V ⊆ f ⁻¹' U`,
+this is the induced map `Γ(Y, U) ⟶ Γ(X, V)`. -/
+abbrev Hom.appLe {X Y : Scheme} (f : X ⟶ Y) {V : Opens X.carrier} {U : Opens Y.carrier}
+    (e : V ≤ (Opens.map f.1.base).obj U) : Y.presheaf.obj (op U) ⟶ X.presheaf.obj (op V) :=
+  f.1.c.app (op U) ≫ X.presheaf.map (homOfLE e).op
+#align algebraic_geometry.Scheme.hom.app_le AlgebraicGeometry.Scheme.Hom.appLe
+
+/-- The spectrum of a commutative ring, as a scheme.
+-/
+def specObj (R : CommRingCat) : Scheme where
+  local_affine _ := ⟨⟨⊤, trivial⟩, R, ⟨(Spec.toLocallyRingedSpace.obj (op R)).restrictTopIso⟩⟩
+  toLocallyRingedSpace := Spec.locallyRingedSpaceObj R
+#align algebraic_geometry.Scheme.Spec_obj AlgebraicGeometry.Scheme.specObj
+
+@[simp]
+theorem specObj_toLocallyRingedSpace (R : CommRingCat) :
+    (specObj R).toLocallyRingedSpace = Spec.locallyRingedSpaceObj R :=
+  rfl
+#align algebraic_geometry.Scheme.Spec_obj_to_LocallyRingedSpace AlgebraicGeometry.Scheme.specObj_toLocallyRingedSpace
+
+/-- The induced map of a ring homomorphism on the ring spectra, as a morphism of schemes.
+-/
+def specMap {R S : CommRingCat} (f : R ⟶ S) : specObj S ⟶ specObj R :=
+  (Spec.locallyRingedSpaceMap f : Spec.locallyRingedSpaceObj S ⟶ Spec.locallyRingedSpaceObj R)
+#align algebraic_geometry.Scheme.Spec_map AlgebraicGeometry.Scheme.specMap
+
+@[simp]
+theorem specMap_id (R : CommRingCat) : specMap (𝟙 R) = 𝟙 (specObj R) :=
+  Spec.locallyRingedSpaceMap_id R
+#align algebraic_geometry.Scheme.Spec_map_id AlgebraicGeometry.Scheme.specMap_id
+
+theorem specMap_comp {R S T : CommRingCat} (f : R ⟶ S) (g : S ⟶ T) :
+    specMap (f ≫ g) = specMap g ≫ specMap f :=
+  Spec.locallyRingedSpaceMap_comp f g
+#align algebraic_geometry.Scheme.Spec_map_comp AlgebraicGeometry.Scheme.specMap_comp
+
+/-- The spectrum, as a contravariant functor from commutative rings to schemes.
+-/
+def Spec : CommRingCatᵒᵖ ⥤ Scheme where
```

Comment:
```suggestion
-- porting note: removed @[simps]
def Spec : CommRingCatᵒᵖ ⥤ Scheme where
```
Even if you backport, anything that's not in the SHA at the top of the file should be a porting note. The notes can be removed if/when the backport is merged.